### PR TITLE
WIP Initial Code Camp post

### PR DESCRIPTION
**Pull request by rstiles**

### CODE REVIEW

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+tags: code camp, engineering
+author: Rachel Stiles
+layout: article
+image:
+  feature:
+  alt:
+---
+
+Every year, Signavio's hard-working engineering department splits into groups to work on a variety of projects that they create from brainstorming. They present these projects at the end of the yearly Code Camp in front of the entire company. The best project is selected by vote via the company, and it becomes the newest project that our team works on.
+
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: "it becomes the newest project that our team works on." actually: "its team receives the _Code Camp Excellence Award_". We will continue all projects that make sense, so typically the "low hanging fruit" projects.

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+This year, we had 10 groups working on a variety of projects, from a private, in-house Slack services to a re-design of our Editor's shape menu. Here is a quick summary of each.
+
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: "Editor's shape menu" -> we need to be more specific here, because readers won't know what it is. Perhaps: "the shape menu [link to docs] of our BPMN editor" or similar

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+## Mix Me
+
+Progressive web app--React on front-end, Note.js on back-end
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: "Note.js" -> "Node.js": https://nodejs.org/en/

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+Takes Wednesday tradition and extends it to the entire week--a way to maintain the “everybody knows everybody else” culture of Signavio even as the company continues to grow. It's a way to ensure we still know who we're working with, even when we end up with hundreds of employees.
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: What's "Wednesday tradition"? We sometimes call it "funny Friday"

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+author: Rachel Stiles
+layout: article
+image:
+  feature:
+  alt:
+---
+
+Every year, Signavio's hard-working engineering department splits into groups to work on a variety of projects that they create from brainstorming. They present these projects at the end of the yearly Code Camp in front of the entire company. The best project is selected by vote via the company, and it becomes the newest project that our team works on.
+
+This year, we had 10 groups working on a variety of projects, from a private, in-house Slack services to a re-design of our Editor's shape menu. Here is a quick summary of each.
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: "from a private, in-house Slack services" -> perhaps just call it "Slack bots for our different products" or something.

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+## Oops, did I do that?
+
+Register bugs, see who last worked on the code, find out where the bugs are in the code
+
+Uses Git, slava.jack
+
+##Confluence BPMN Diagram Plugin
+
+A plugin that allows the user to integrate Confluence with our Signavio suite
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: * integrate any diagram type
* embed interactive decision simulation

**Comment by TimKam** [comment]
Some ideas:
* Ask Adam for photos
* Add a disclaimer that these ideas might not make it into production

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
+---
+title: "Code Camp Projects 2017"
+description: ""
+tags: code camp
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: perhaps add: ", off-site, fun projects" or similar

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+author: Rachel Stiles
+layout: article
+image:
+  feature: /2017/codecamp.jpg
+  alt: ""
+---
+
+Every year, Signavio's hard-working engineering department takes a break from their pursuit of BPM excellence to participate in Code Camp, a yearly hackathon.
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: _yearly hackathon_: _yearly internal hackathon_ or similar. Should be clear it's only for the company. Also, you could mention that not only devs are part of code camp, although I'm not sure if it's important to explain this.

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+They divide into groups, brainstorm ideas, and get to work creating a variety of projects.
+At the end of Code Camp, they present their finished projects to the entire company.
+The best project is selected by vote via the company, and it becomes the newest project that our team works on.
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: _and it becomes the newest project that our team works on._ -> _and it wins the prestigious Code Camp Excellence Award._ ;-) Add: _We later evaluate all projects and continue the ones that can create value for the company._ or similar.

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+
+This year, we had 10 groups working on a variety of projects, from a Slackbot integration with our Workflow Accelerator to a re-design of our Editor's shape menu. Here is a quick summary of each.
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: _Editor's shape menu_ -> _of our process editor's shape menu_ I wouldn't capitalize "Editor" here

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+
+**Note** These are simply projects our engineers worked on for challenge and fun--they may never actually make it into production.
+
+
+##A better Shape Menu
+
+When the Editor first launched, the menu was revolutionary. It was easy to use, it allowed the user to see all tool options in a clear way, and it was a pleasure to navigate.
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: _When the Editor first launched, the menu was revolutionary._ -> _When our process modeling tool first launched, it's shape menu was revolutionary._ or similar

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+As with all good ideas, the design was eventually adopted industry-wide, and is now not so remarkable.
+This group looked at the menu and though--how can we improve this? How can we create a new menu that is every bit as intuitive and astounding as the original version was when it first released?
+
+The solution they came up with includes bigger icons, a less crammed menu, and adding a touch function.
+The menu this group designed also has two modes--the quick mode, which has bigger icons and a more user-friendly menu designed with touch screens in mind; and what they call the rocket mode, which text-based and designed with modeling speed in mind.
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: _which text-based_ -> _which is text based_

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+This year, we had 10 groups working on a variety of projects, from a Slackbot integration with our Workflow Accelerator to a re-design of our Editor's shape menu. Here is a quick summary of each.
+
+**Note** These are simply projects our engineers worked on for challenge and fun--they may never actually make it into production.
+
+
+##A better Shape Menu
+
+When the Editor first launched, the menu was revolutionary. It was easy to use, it allowed the user to see all tool options in a clear way, and it was a pleasure to navigate.
+As with all good ideas, the design was eventually adopted industry-wide, and is now not so remarkable.
+This group looked at the menu and though--how can we improve this? How can we create a new menu that is every bit as intuitive and astounding as the original version was when it first released?
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: _and though_ -> _and thought_. I'd remove _how can we improve this?_ because the next question implies this as well.
_as the original version was when it first released?_ -> _as the original version was when it **was** first released?_

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+The menu this group designed also has two modes--the quick mode, which has bigger icons and a more user-friendly menu designed with touch screens in mind; and what they call the rocket mode, which text-based and designed with modeling speed in mind.
+According to the group, using the rocket mode can cut modeling time in half.
+
+## Mix Me
+
+At Signavio, we have a monthly tradition called Mix & Match Lunch where we randomly divide into groups of people we don't usually interact with, and go to lunch.
+It's a way to ensure we know everyone we're working with, even if we don't see them every day.
+This app is a way to extend that tradition to the entire week, maintaining our company culture even as we grow.
+
+Mix Me is a progressive web app, utilizing REST and Note.js.
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: _Note.js_ -> _Node.js_, see_ https://nodejs.org
Instead of mentioning REST, I'd mention React.

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+A user logs onto the Mix Me webpage, enters their Signavio email and is randomly paired up with a group of four other colleagues, a random restaurant near Signavio HQ and a suggested meeting place.
+
+## SWAccelerator Talk - Alexa + Workflow Accelerator
+
+This project focused on an actual use-case that involves integrating Alexa with our Workflow Accelerator, utilizing Note.js.
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: _Note.js_ -> _Node.js_
I wouldn't call it an _actual use-case_, rather an _entertaining demo scenario_ :-)

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+
+Say that a company's social media team wants to allow employees to send out tweets, but is unsure how to run a quality control process that would prevent any gaffes that might embarrass the company.
+Using Workflow Accelerator, you would be able to integrate it with Alexa.
+Simply dictate the tweet to Alexa, who will compare the text of the tweet to its library of acceptable syntax and phrasing.
+
+If the tweet is approved, the Workflow Accelerator will send an email to the social media team alerting them to the tweet and allowing them to either approve it or reject it.
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: _the Workflow Accelerator_ -> _Workflow Accelerator_ (we don't use articles in front of product names)

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+If Alexa determines the tweet is not acceptable, it will reject it and ask the user to try again.
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: Perhaps worth mentioning: _In addition, Alexa provides feedback on how the tweet can be improved. To realize this, the team integrated Workflow Accelerator with [Alex](https://github.com/wooorm/alex)._

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+
+While Alexa serves as a filter for content, the ultimate authority that approves or disapproves tweets is the company's social media team.
+
+## Fast. Frictionless. Fun.
+
+Ever wished Signavio had an app version?
+Not to create diagrams with, but an app that would allow you to see the core information of your projects, write down notes, send feedback, etc. all while you're on the go?
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: _etc._ -> _and more_?

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+Well, so did we!
+
+This group focused on making a mobile version of our Workflow Editor for Android.
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: _Workflow Editor_ -> _Process Manager_

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 20]
```diff
@@ -0,0 +1,116 @@
...
+The main challenge for this group was making things lighter and faster than the desktop version, with comments and share capabilities as well as a way to check in on project details while on the go.
+
+The resulting app is a mobile version of our Workflow Editor, minus the workflow capability.
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: _Workflow Editor_ -> _Process Manager_
_workflow capability_ -> _process modeling capability_

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 21]
```diff
@@ -0,0 +1,116 @@
...
+This group focused on making the app as light and fast as possible, ensure it is optimized for mobile use.
+Users can comment, share snapshots of process diagram models, make notes and similar tasks, which makes this mobile app a sort of one stop shop for info about any process models you may be creating.
+
+Mobile app--dashboard for comments, page views, menu for people to browse folders for the editor
+Mobile editor with feed for comments and a snapshot of the models--but not the actual models themselves--just like a one stop shop for info about it
+Android for now
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: You left some notes here: I think they can be removed.

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 22]
```diff
@@ -0,0 +1,116 @@
...
+
+The solution our engineers came up with is a traffic light that alerts the room when things are getting a little loud.
+Using Raspberry Pi and a microphone, hooked up to a physical light and using a program our devs wrote to make it all talk to each other.
+The program is set to detect a certain volume level, and to announce (via a green, yellow or red light) when changes in that level are detected.
+
+Helpfully, the app also tracks and graphs noise levels over time, allowing the admin to see what the noisiest times of the day are and what causes certain spikes in noise.
+
+## Slacknavio
+
+Sometimes, when using Signavio, a user wants to check something small--conversations they're a part of, perhaps, or maybe just double-check the tasks they are assigned--but they don't want to have to log into Workflow Accelerator or the Collaboration Hub just for a quick look.
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: _the Collaboration Hub_ -> _Collaboration Hub_

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 23]
```diff
@@ -0,0 +1,116 @@
...
+
+That's where the Slacknavio bot comes in.
+This is a Slack bot that allows users to pull basic information from Workflow Accelerator or the Collaboration Hub instantly.
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: _the Collaboration Hub_ -> _Collaboration Hub_

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 24]
```diff
@@ -0,0 +1,116 @@
...
+
+Created with [Go] (https://golang.org/), the Slacknavio bot can post updates about new diagrams in the Collaboration Hub, new personal tasks assigned to users in the Accelerator, notify users of new comments in conversation threads they are included in, as well as search the Signavio repository and share daily status updates on things such as workflow processes.
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: _the Collaboration Hub_ -> _Collaboration Hub_

Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: _the Accelerator_ -> _Workflow Accelerator_

Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: _on things such as workflow processes_ -> _on things such as workflows and cases_

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 25]
```diff
@@ -0,0 +1,116 @@
...
+
+## Workflow4Slack
+
+Let's say you want to more easily track what is going on in the Workflow Accelerator, but you don't want to wait for your colleagues to get the email notifications.
+This project integrates Slack with the Workflow Accelerator to notify users when changes have been made in their workspace.
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: _the Workflow Accelerator_ -> _Workflow Accelerator_

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 26]
```diff
@@ -0,0 +1,116 @@
...
+Sometimes, when using Signavio, a user wants to check something small--conversations they're a part of, perhaps, or maybe just double-check the tasks they are assigned--but they don't want to have to log into Workflow Accelerator or the Collaboration Hub just for a quick look.
+
+That's where the Slacknavio bot comes in.
+This is a Slack bot that allows users to pull basic information from Workflow Accelerator or the Collaboration Hub instantly.
+
+Created with [Go] (https://golang.org/), the Slacknavio bot can post updates about new diagrams in the Collaboration Hub, new personal tasks assigned to users in the Accelerator, notify users of new comments in conversation threads they are included in, as well as search the Signavio repository and share daily status updates on things such as workflow processes.
+
+## Workflow4Slack
+
+Let's say you want to more easily track what is going on in the Workflow Accelerator, but you don't want to wait for your colleagues to get the email notifications.
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: _the Workflow Accelerator_ -> _Workflow Accelerator_

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 27]
```diff
@@ -0,0 +1,116 @@
...
+Made changes to a process and need some feedback from your colleagues?
+You can send them a notification from Workflow Accelerator via Slack.
+It's a good option for those times when email simply takes too long.
+
+## Oops, did I do that?
+When a bug is discovered, it can sometimes be a challenge to locate where exactly it is in the code.
+When you're collaborating with many people on a team, it can also be a bit tricky to find a solution, because it's not always easy to figure out how the bug happened or what caused it or if it was caused by a change someone else made.
+
+This program seeks to make the de-bugging process a little easier by creating a log.
+Users can register bugs as they find them, along with the location of the bug, as well as find out who last worked on that part of the code.
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: I think the exact use case is a bit different: as soon as a bug occurs, we (meaning devs and support engineers, not the users) get an automatic notification and can track the exact place in the code where the bug is caused and the system tells use the name of the engineer who is most likely to know how to fix it. This means in the best case, we can fix a bug before the users even realize that it exists. Perhaps as Christian Schwarz about it.

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 28]
```diff
@@ -0,0 +1,116 @@
...
+
+
+##Confluence BPMN Diagram Plugin
+
+A plugin that allows the user to integrate Confluence with our Signavio suite
+
+##Gmail Signature Updates
+
+Keeping the company email signatures up to date is always a challenge, especially when the company is growing as quickly as Signavio.
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: You could elaborate in 1-2 sentences that currently, Marketing is sending us a mail with a new signature every couple of weeks and most people are too lazy to update it ;-)

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 29]
```diff
@@ -0,0 +1,116 @@
...
+When a bug is discovered, it can sometimes be a challenge to locate where exactly it is in the code.
+When you're collaborating with many people on a team, it can also be a bit tricky to find a solution, because it's not always easy to figure out how the bug happened or what caused it or if it was caused by a change someone else made.
+
+This program seeks to make the de-bugging process a little easier by creating a log.
+Users can register bugs as they find them, along with the location of the bug, as well as find out who last worked on that part of the code.
+
+
+##Confluence BPMN Diagram Plugin
+
+A plugin that allows the user to integrate Confluence with our Signavio suite
```
Comment (TimKam) [inline _posts/2017-10-07-code-camp-projects-2017.md]: Elaborate!
- display diagrams right alongside your good old confluence content
- embed the decision assistant
- Ask @SilentGert or @nfaupel for a screenshot of the decision assistant :-)

File: _posts/2017-10-07-code-camp-projects-2017.md (hunk 7 above)
Comment (nfaupel) [reply _posts/2017-10-07-code-camp-projects-2017.md]: hackathon would also fit.

File: _posts/2017-10-07-code-camp-projects-2017.md (hunk 21 above)
Comment (rstiles) [reply _posts/2017-10-07-code-camp-projects-2017.md]: 😅

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 30]
```diff
@@ -0,0 +1,120 @@
...
+tags: code camp, hackathon, off-site
+author: Rachel Stiles
+layout: article
+image:
+  feature: /2017/code-camp.jpg
+  alt: "Signavio co-CTO Willi Tscheschner kicking off Code Camp 2017"
+---
+
+Every year, Signavio's hard-working engineering department takes a break from their pursuit of BPM excellence to participate in Code Camp, our yearly internal hackathon.
+They divide into groups, brainstorm ideas, and get to work creating a variety of projects (the other departments, meanwhile, spend time in workshops learning about different subjects).
```
Comment (steffektif) [inline _posts/2017-10-07-code-camp-projects-2017.md]: Probably you could rephrase this to: While the other departments ..., our engineers divide into groups.. 

Not 100% sure if it's better, probably choose another word than engineer then.

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 31]
```diff
@@ -0,0 +1,120 @@
...
+Helpfully, the app also tracks and graphs noise levels over time, allowing the admin to see what the noisiest times of the day are and what causes certain spikes in noise.
+
+## Slacknavio
+
+Sometimes, when using Signavio, a user wants to check something small--conversations they're a part of, perhaps, or maybe just double-check the tasks they are assigned--but they don't want to have to log into Workflow Accelerator or Collaboration Hub just for a quick look.
+
+That's where the Slacknavio bot comes in.
+This is a Slack bot that allows users to pull basic information from Workflow Accelerator or Collaboration Hub instantly.
+
+Created with [Go] (https://golang.org/), the Slacknavio bot can post updates about new diagrams in Collaboration Hub, new personal tasks assigned to users in Workflow Accelerator, notify users of new comments in conversation threads they are included in, as well as search the Signavio repository and share daily status updates on things such as workflows and cases.
```
Comment (steffektif) [inline _posts/2017-10-07-code-camp-projects-2017.md]: Well, in the end the search thingy didn't work out. So this wouldn't be true, but for making it look a lil more nice you could of cause keep it.

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 32]
```diff
@@ -0,0 +1,120 @@
...
+
+## Workflow4Slack
+
+Let's say you want to more easily track what is going on in Workflow Accelerator, but you don't want to wait for your colleagues to get the email notifications.
+This project integrates Slack with Workflow Accelerator to notify users when changes have been made in their workspace.
+
+Want to notify a user that they've been assigned a new task? Workflow4Slack will ping them.
```
Comment (steffektif) [inline _posts/2017-10-07-code-camp-projects-2017.md]: This team also did more than notifying people in a workflow, but also let them fill forms (from Signavio Workflow Accelerator user tasks) within Slack. This way users don't even need to open up SWAG'ccelerator, log in and fill the form. But could easily do this within Slack.

File: _posts/2017-10-07-code-camp-projects-2017.md [hunk 33]
```diff
@@ -0,0 +1,120 @@
...
+Made changes to a process and need some feedback from your colleagues?
+You can send them a notification from Workflow Accelerator via Slack.
```
Comment (steffektif) [inline _posts/2017-10-07-code-camp-projects-2017.md]: ... via Slack. Even filling a Usertasks form within Slack is possible.
